File: src/Reaper.hs
{-# LANGUAGE QuasiQuotes #-}
{-# LANGUAGE TemplateHaskell #-}

module Reaper (subreaper) where

import qualified Language.C.Inline as C

C.include "<sys/prctl.h>"
C.include "<errno.h>"

subreaper :: IO Int
subreaper = fromIntegral <$> [C.block| int
    { 
        return prctl(PR_SET_CHILD_SUBREAPER, 1, 0, 0, 0) < 0 ? errno : 0;
    } |]

File: src/Reaper.c

#include <sys/prctl.h>

#include <errno.h>

int inline_c_Reaper_0_9c414a0e9697bdfa11134cc5646b6baaa1616203() {
 
        return prctl(PR_SET_CHILD_SUBREAPER, 1, 0, 0, 0) < 0 ? errno : 0;
    
}

